index.js

```diff
@@ -1,18 +1,23 @@
-var bufferEqual = require('buffer-equal-constant-time');
-var Buffer = require('safe-buffer').Buffer;
+var Buffer = require('buffer').Buffer;
 var crypto = require('crypto');
 var formatEcdsa = require('ecdsa-sig-formatter');
 var util = require('util');
 
 var MSG_INVALID_ALGORITHM = '"%s" is not a valid algorithm.\n  Supported algorithms are:\n  "HS256", "HS384", "HS512", "RS256", "RS384", "RS512", "PS256", "PS384", "PS512", "ES256", "ES384", "ES512" and "none".'
-var MSG_INVALID_SECRET = 'secret must be a string or buffer';
-var MSG_INVALID_VERIFIER_KEY = 'key must be a string or a buffer';
+var MSG_INVALID_SECRET = 'secret must be a string or buffer or a KeyObject';
+var MSG_INVALID_VERIFIER_KEY = 'key must be a string or a buffer or a KeyObject';
 var MSG_INVALID_SIGNER_KEY = 'key must be a string, a buffer or an object';
 
-var supportsKeyObjects = typeof crypto.createPublicKey === 'function';
-if (supportsKeyObjects) {
-  MSG_INVALID_VERIFIER_KEY += ' or a KeyObject';
-  MSG_INVALID_SECRET += 'or a KeyObject';
+/**
+ * @param {Uint8Array} a
+ * @param {Uint8Array} b
+ */
+function timingSafeEqual(a, b) {
+  if (a.length !== b.length) return false
+
+  let c = 0;
+  for (let i = 0, len = a.length; i < len; i++) c |= a[i] ^ b[i];
+  return !c;
 }
```

Comment:
Should probably use [crypto.timingSafeEqual](https://nodejs.org/api/crypto.html#cryptotimingsafeequala-b) instead.

Reply:
I don't think the time has come for that yet

- node's webcrypto isn't stable until v20.x
- node's first LTS release with a globally exposed webcrypto isn't even out yet (v20.x due in october)
- portability across runtimes is not a goal for this module, hence its `node-` prefix
- node's webcrypto doesn't accept KeyObject instances, time would be wasted waiting for promises to import the key material as CryptoKey
- WebCryptoAPI is entirely promise-based and majority of developers use node-jws/node-jwa (through jsonwebtoken) in a synchronous way.

Requiring node 20.x to be used is just premature given how slowly people migrate off of EOL node versions and the popularity of the modules that depend on node-jwa.

Reply:
hmm, you are right. it maybe would be a too breaking change.
but maybe that don't stop this from being conditionally dependent on using whatever is available? if it has globalThis.crypto - then use that otherwise conditionally import node:crypto

```js
const crypto = globalThis.crypto || await import('node:crypto')
```  
but again. maybe this is too braking also...
maybe this can be revisited later

Reply:
Other modules exist as an option for JWS[TEAK] with portability across runtimes as a goal. 

I'm not saying node-jwa/node-jws/jsonwebtoken won't get there eventually but it would be a big distraction to the ecosystem if all of a sudden async promise based API was the only option. With a module this old and popular updating the API is way too big a deal because you're not only dealing with legacy use who just expects major breaking changes to be no-ops for most, but the vast majority of online material for jsonwebtoken would instantly become obsolete with no way of updating it all.